### PR TITLE
[react-datepicker] Change previous/nextMonthButtonLabel to include Re…

### PR DESCRIPTION
**Pull request by ruiyu2000**
…actNode

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/Hacker0x01/react-datepicker/pull/2124/files>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/react-datepicker/index.d.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export interface ReactDatePickerProps {
     minTime?: Date;
     monthsShown?: number;
     name?: string;
-    nextMonthButtonLabel?: string;
+    nextMonthButtonLabel?: string| React.ReactNode;
```
Comment (amcasey) [inline types/react-datepicker/index.d.ts:83]: Missed a space?

**Review by amcasey — CHANGES_REQUESTED** [review]
Why would this apply to months but not to years?

File: types/react-datepicker/index.d.ts (hunk 1 above)
Comment (ruiyu2000) [reply types/react-datepicker/index.d.ts:83]:
```suggestion
    nextMonthButtonLabel?: string | React.ReactNode;
```

**Comment by ruiyu2000** [comment]
> Why would this apply to months but not to years?

https://github.com/Hacker0x01/react-datepicker/blob/master/src/index.jsx#L243
`
previousYearButtonLabel: PropTypes.string,
nextYearButtonLabel: PropTypes.string,
`

**Review by amcasey — APPROVED** [review]

**Comment by ruiyu2000** [comment]
Ready to merge

**Pull request merged**